backend/dataInterface/mongoDB.js

```diff
@@ -90,56 +284,63 @@ module.exports.createUser = async (user) => {
     let emailExist = await collection.findOne({ email: user.email });
     if (emailExist) {
       return { Error: "Email already exists" };
-    }
-    else{
+    } else {
       let hashedPassword = await bcrypt.hash(user.password, 10);
       let vettedUser = {
         username: user.username,
         email: user.email,
         password: hashedPassword,
         firstName: user.firstName,
-        lastName: user.lastName
+        lastName: user.lastName,
       };
       const result = await collection.insertOne(vettedUser);
-      return { newObjectId: result.insertedId, message: `User created! ID: ${result.insertedId}`, pass: hashedPassword };
+      return {
+        newObjectId: result.insertedId,
+        message: `User created! ID: ${result.insertedId}`,
+        pass: hashedPassword,
+      };
     }
   } catch (err) {
-    return {Error: 'Failed to Create'};
+    return { Error: "Failed to Create User" };
   } finally {
     await client.close();
   }
-}
+};
 
-// 
-module.exports.signIn = async(user) =>{
-  try{
+//user login
+module.exports.signIn = async (user) => {
+  try {
     await client.connect();
     const db = client.db(databaseName);
     const collection = db.collection(users_Collection);
-    const userExistEmail = await collection.findOne({email: user.usernameOrEmail})
-    const userExistUsername = await collection.findOne({username: user.usernameOrEmail})
-    if(userExistEmail){
-      if(await bcrypt.compare(user.password, userExistEmail.password )){
-        return {username: userExistEmail.username, email: userExistEmail.email}
+    const userExistEmail = await collection.findOne({
+      email: user.usernameOrEmail,
+    });
+    const userExistUsername = await collection.findOne({
+      username: user.usernameOrEmail,
+    });
+    if (userExistEmail) {
+      if (await bcrypt.compare(user.password, userExistEmail.password)) {
+        return {
+          username: userExistEmail.username,
+          email: userExistEmail.email,
+        };
       }
     }
-    if(userExistUsername){
-      if(await bcrypt.compare(user.password, userExistUsername.password)){
-        return {username: userExistUsername.username, email: userExistUsername.email}
+    if (userExistUsername) {
+      if (await bcrypt.compare(user.password, userExistUsername.password)) {
+        return {
+          username: userExistUsername.username,
+          email: userExistUsername.email,
+        };
```

Comment:
These two conditions can be combined:
```
if (userExistEmail || userExistUsername) {
      if (await bcrypt.compare(user.password, userExistEmail.password)) {
        return {
          username: userExistEmail.username,
          email: userExistEmail.email,
        };
      }
    }
```
There should also be an `else` block that throws an error, in case the user cannot be found by their email or username.